src/ui/public/crypto/sha256.js

```diff
@@ -0,0 +1,194 @@
+// ported from https://github.com/spalger/sha.js/blob/6557630d508873e262e94bff70c50bdd797c1df7/sha256.js
```

Comment:
Topic for discussion: are there any non-cryptographic hash functions we can use that emphasize uniqueness and speed?

I did some googling and found this StackExchange topic on ["Which hashing algorithm is best for uniqueness and speed?"](http://programmers.stackexchange.com/questions/49550/which-hashing-algorithm-is-best-for-uniqueness-and-speed). It seemed to recommend using [Murmur](http://stackoverflow.com/questions/11899616/murmurhash-what-is-it), which has a couple implementations on NPM:

* https://www.npmjs.com/package/murmur
* https://www.npmjs.com/package/murmurhash-js

The reason I ask about a different hashing algo is because we really don't need the cryptographic nature of SHA, so if we remove that from the equation we can get something faster and with a simpler implementation.